src/Hypsography/Hypsography.jl

```diff
@@ -2,10 +2,174 @@ module Hypsography
 
```

Comment:
This combines `adaptspace.jl` and `terrainadapt.jl`. `Adaption` now takes the surface elevation as an argument. @simonbyrne It seems like this would change the ClimaAtmos API; this is fine with me. This is functionally (i.e. w.r.t mesh metrics) the same as the existing system.

Reply:
Yes, it is breaking: the main question is what code is using this API? I might be able to add some deprecations if need be.